Core/MIPS/x86/CompLoadStore.cpp

```diff
@@ -429,6 +429,7 @@ namespace MIPSComp {
 			skipStore = J_CC(CC_NE);
 
 			CompITypeMemWrite(op, 32, safeMemFuncs.writeU32);
+			gpr.MapReg(rt, true, true);
```

Comment:
Wait, this looks wrong.  You can't map in a conditional, it could corrupt other regs from the perspective of the other condition.  This needs to happen before the J_CC?

-[Unknown]

Reply:
Since the reg is only written to in both branches, I actually think it should be fine here?

Reply:
But hm, I'll clean it up.

Reply:
No, because another reg might spill.  If it only spills in one conditional, then that other reg is corrupted.

-[Unknown]

Reply:
A problem is that CompITypeMemWrite itself locks and unlocks regs, so I can't reliably do it that way, can I?

Reply:
Hm, I think the actual bug is in CompITypeMemWrite, in the gpr.R(rt).IsImm() case, since normally we'd end up with rt being mapped coming out of it.

I'll fix that.

Reply:
Well, not so much a bug as CompITypeMemWrite just not being designed to leave rt writable, fixable with a flag.